src/Microsoft.Extensions.ProjectModel.MsBuild.Sources/MsBuildProjectContextBuilder.cs

```diff
@@ -37,7 +42,7 @@ public IProjectContext Build()
                     "msbuild",
                     _projectPath,
                     $"/t:EvaluateProjectInfoForCodeGeneration", 
-                    $"/p:TargetFramework={_targetFramework.GetShortFolderName()};OutputFile={tmpFile}"
+                    $"/p:OutputFile={tmpFile};CodeGenerationTargetLocation={_targetLocation}"
```

Comment:
I tried following this code path but I think I missed something. From looking at how the project context builder is constructed the target location looks like it points to the users project directory. I'd think it'd be pointing to the NuGet cache.